### PR TITLE
Drop support for EOL Python 2.6-3.5

### PR DESCRIPTION
**Pull request by hugovk**
Replaces and closes https://github.com/lericson/pylibmc/pull/256.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{26,27,33,34,35,36,37,38}
+envlist = py{35,36,37,38}
```
Comment (hroncok) [inline tox.ini]: 3.8 in here...

File: setup.py [hunk 2]
```diff
@@ -112,17 +109,13 @@ def append_env(L, e):
...
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.6',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.2',
-        'Programming Language :: Python :: 3.3',
-        'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
```
Comment (hroncok) [inline setup.py]: ...but no 3.8 in here.

Comment (hugovk) [reply setup.py]: I had left it out as it's also in PR https://github.com/lericson/pylibmc/pull/257, but I think it makes sense to put it here too. Updated!

**Comment by lericson** [comment]
Let's do that!

**Comment by hugovk** [comment]
PR updated to drop EOL Python 3.5.

File: src/_pylibmcmodule.c [hunk 3]
```diff
@@ -600,11 +580,7 @@ static PyObject *_PylibMC_parse_memcached_value(PylibMC_Client *self,
     if (self->native_deserialization) {
         retval = _PylibMC_deserialize_native(self, NULL, value, size, flags);
     } else {
-#if PY_MAJOR_VERSION >= 3
         retval = PyObject_CallMethod((PyObject *)self, "deserialize", "y#I", value, size, (unsigned int) flags);
-#else
-        retval = PyObject_CallMethod((PyObject *)self, "deserialize", "s#I", value, size, (unsigned int) flags);
-#endif
```
Comment (lericson) [inline src/_pylibmcmodule.c:607]: Oh lord this is turning me on

File: src/_pylibmcmodule.c [hunk 4]
```diff
@@ -1257,24 +1226,11 @@ static int _PylibMC_serialize_native(PylibMC_Client *self, PyObject *value_obj,
...
-        store_flags |= PYLIBMC_FLAG_INTEGER;
-        PyObject* tmp = PyNumber_Int(value_obj);
-        store_val = PyObject_Bytes(tmp);
-        Py_DECREF(tmp);
-    } else if (PyLong_Check(value_obj)) {
-        store_flags |= PYLIBMC_FLAG_LONG;
-        PyObject* tmp = PyNumber_Long(value_obj);
-        store_val = PyObject_Bytes(tmp);
-        Py_DECREF(tmp);
-#endif
```
Comment (lericson) [inline src/_pylibmcmodule.c:1277]: *wipes sweat from forehead* Yessssss

File: src/pylibmc/client.py [hunk 5]
```diff
@@ -139,18 +139,18 @@ def __init__(self, servers, behaviors=None, binary=False,
         """
         self.binary = binary
         self.addresses = list(servers)
-        super(Client, self).__init__(servers=translate_server_specs(servers),
+        super().__init__(servers=translate_server_specs(servers),
```
Comment (lericson) [inline src/pylibmc/client.py:142]: Alignment is wrong for the lines below

File: src/pylibmc/client.py [hunk 6]
```diff
@@ -139,18 +139,18 @@ def __init__(self, servers, behaviors=None, binary=False,
...
                                      binary=binary,
                                      username=username, password=password,
                                      behaviors=_behaviors_numeric(behaviors))
 
     def __repr__(self):
-        return "%s(%r, binary=%r)" % (self.__class__.__name__,
+        return "{}({!r}, binary={!r})".format(self.__class__.__name__,
```
Comment (lericson) [inline src/pylibmc/client.py:148]: Alignment is wrong for next line

File: src/pylibmc/client.py [hunk 7]
```diff
@@ -139,18 +139,18 @@ def __init__(self, servers, behaviors=None, binary=False,
...
                                       self.addresses, self.binary)
 
     def __str__(self):
         addrs = ", ".join(map(str, self.addresses))
-        return "<%s for %s, binary=%r>" % (self.__class__.__name__,
+        return "<{} for {}, binary={!r}>".format(self.__class__.__name__,
```
Comment (lericson) [inline src/pylibmc/client.py:153]: Alignment

**Review by lericson — CHANGES_REQUESTED** [review]
Wow, massively good upgrades all around. Probably should do a once-over for the C code also, a lot of type checking hijinks is really for py2+py3 support I think.

File: bin/runbench.py [hunk 8]
```diff
@@ -103,9 +101,9 @@ def bench_incr_decr(mc, key):
 
 
 def multi_pairs(n, *keys):
-    d = dict((b'%s%d' % (k, i), b'data%s%d' % (k, i))
+    d = {b'%s%d' % (k, i): b'data%s%d' % (k, i)
```
[the remark this answers is not in the document]
Comment (hugovk) [reply bin/runbench.py:104]: pyupgrade is intentionally timid and doesn't rewrite some percent formatter statements to be on the safe side.

https://github.com/asottile/pyupgrade#f-strings

[the remark this answers is not in the document]
Comment (lericson) [reply bin/runbench.py:104]: Ah, it's must be because it's a bytes literal. Eh, I'm sure it's fine.

File: src/pylibmc/client.py (hunk 5 above)
Comment (lericson) [reply src/pylibmc/client.py:142]: Hm it should align with the opening `(` of line 142

Comment (lericson) [reply src/pylibmc/client.py:142]: Oh it does now, it was just displayed weirdly

**Review by lericson — APPROVED** [review]
Great!

**Pull request merged**